### PR TITLE
Convert rename-related pages to Foundation

### LINKED ISSUE #972: Convert rename.tt and rename/swap.tt to Foundation

### LINKED ISSUE #1504: [Safari+JS] problem with input focus on /rename page
If I visit dreamwidth.org/rename and try to input text into the "Rename token" text field, the focus immediately jumps up to the "Desired username" field.  Using control-click to paste text is the only workaround I've figured out so far.

I'm only able to reproduce this in Safari.  If I use Firefox or Chrome, the focus doesn't jump away.  I haven't checked to see if earlier versions of Safari are affected, but the same thing seems to happen in Safari on my iPad running iOS 8.3 as well.

There's no special JavaScript being loaded on the page, just the default site-wide stuff.

### PR DESCRIPTION
**Pull request by kareila**
This fixes #972 by converting the two pages named there, but also includes all the rename admin pages as well, since most of them used the same error handling mechanism, which I updated to use DW::FormErrors.

This is lightly tested; I checked the default display of most of the pages, and generated a few error messages to make sure they appeared as expected.

This series of changes also fixes #1504, which I uncovered in the process of testing.

### CODE REVIEW

File: cgi-bin/DW/RenameToken.pm [hunk 1]
```diff
@@ -109,7 +109,7 @@ Verifies if this could be a valid format for the rename token. Checks length and
 sub valid_format {
     my ( $class, %opts ) = @_;
 
-    my $string = uc $opts{string};
+    my $string = defined $opts{string} ? uc $opts{string} : '';
```
Comment (zorkian) [inline cgi-bin/DW/RenameToken.pm]: Or,

```
my $string = uc($opts{string} // '');
```

File: cgi-bin/DW/User/Rename.pm [hunk 2]
```diff
@@ -57,34 +58,36 @@ Return true if this user can be renamed to the given username
 
 Optional arguments are:
 =item force     => bool, default false
-=item errref    => array ref of errors
+=item errors    => DW::FormErrors object
```
Comment (zorkian) [inline cgi-bin/DW/User/Rename.pm]: Slight preference to `errobj` in these kinds of things, just so it's clear it's an object. (You don't have to change things unless you want, but just a note...)

File: htdocs/scss/pages/rename.scss [hunk 3]
```diff
@@ -25,15 +20,6 @@
...
-}
-
-ul.error-list li { 
-    list-style: disc outside;
-}
-
-p.detail {
-    margin-left: 2em !important;
+#content p.note {
+    font-style: italic; font-size: 80%;
```
Comment (zorkian) [inline htdocs/scss/pages/rename.scss]: I believe the style is to have a newline after semicolons, so they're on separate lines?

File: views/admin/rename_edit.tt [hunk 4]
```diff
@@ -13,16 +13,11 @@ the same terms as Perl itself.  For a copy of the license, please reference
...
 
-[% IF error_list %]
-    <p>Error</p>
-    <ul class='error-list'>
-        [% FOREACH error = error_list %]
-            <li>[% error %] </li>
-        [% END %]
-    </ul>
-[% END %]
+[%- dw.need_res( { group => "foundation" }
```
Comment (zorkian) [inline views/admin/rename_edit.tt]: I can't recall off-hand, but is there really supposed to be no trailing comma here?

Comment (afuna) [inline views/admin/rename_edit.tt]: The comma is optional there in TT.

File: htdocs/scss/pages/rename.scss (hunk 3 above)
Comment (kareila) [inline htdocs/scss/pages/rename.scss]: You're right, I missed that when pasting it from somewhere else.

File: views/admin/rename_edit.tt (hunk 4 above)
Comment (zorkian) [inline views/admin/rename_edit.tt]: Aren't you on vacation? ;)

Comment (kareila) [inline views/admin/rename_edit.tt]: She loves us too much to stay away entirely :wink:

**Comment by zorkian** [comment]
LGTM nothing blocking this from merging, but I won't hit the button given the style nit. Once you're happy with it go ahead and merge etc.

Thanks for doing this! Yay forward progress on our conversions :)

File: views/admin/rename_edit.tt (hunk 4 above)
Comment (afuna) [inline views/admin/rename_edit.tt]: I am not even writing any code! Just... keeping my hand in a bit :)

**Comment by kareila** [comment]
Thanks for the quick review!

File: views/admin/rename_edit.tt (hunk 4 above)
Comment (rahaeli) [inline views/admin/rename_edit.tt]: We are like the Hotel California. You can check out anytime you'd like ...

**Pull request merged**